scheduler/scheduler_test.go

```diff
@@ -107,3 +107,29 @@ func TestScheduler_SyncSchedules(t *testing.T) {
 		})
 	}
 }
+
+func Test_generateName(t *testing.T) {
+	tests := map[string]struct {
+		jobType        k8upv1alpha1.JobType
+		prefix         string
+		expectedPrefix string
+	}{
+		"GivenShortPrefix_WhenGenerate_ThenUseFullPrefix": {
+			jobType:        k8upv1alpha1.ArchiveType,
+			prefix:         "my-schedule",
+			expectedPrefix: "my-schedule-archive-",
+		},
+		"GivenLongPrefix_WhenGenerate_ThenShortenPrefix": {
+			jobType:        k8upv1alpha1.ArchiveType,
+			prefix:         "my-schedule-with-a-really-long-name-that-could-clash-with-max-length",
+			expectedPrefix: "my-schedule-with-a-really-long-name-that-could-archive-",
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			name := generateName(tt.jobType, tt.prefix)
+			assert.Contains(t, name, tt.expectedPrefix)
+			assert.LessOrEqual(t, len(name), 63)
```

Comment:
I suggest to add an assert for the total length as well, i.e. `len(tt.expectedPrefix)+8`.

Reply:
I'm not sure what you mean? the return value is the full name, and that should not exceed total length of 63 chars, which includes prefix, type and suffix

Reply:
As I read the test:
- The test tests that the prefix is as expected
- It tests that the allowed maxima is not exceeded

It does not test however whether the random suffix is actually generated. If the suffix were of length zero, it would still pass the test.
Therefore I suggest a test that checks whether the generated name's length is equal to the length of the prefix plus the expected length of the suffix, i.e. `assert.Equal(t, len(name), len(expectedPrefix)+8`.

Reply:
ah, right